src/Config/ConfigFactory.ts

```diff
@@ -143,17 +161,27 @@ export default class ConfigFactory {
         ),
         facilitatorConfig,
       );
-    } else {
-      const facilitatorConfig: FacilitatorConfig = FacilitatorConfig.fromFile(
-        this.facilitatorConfigPath!,
+    } else if (this.gatewayConfigPath) {
+      const gatewayConfig = GatewayConfig.fromFile(this.gatewayConfigPath);
+
+      this.verifyChainIdInGatewayConfig(gatewayConfig);
+
+      configObj = new Config(
+        GatewayAddresses.fromGatewayConfig(
+          gatewayConfig,
+        ),
+        facilitatorConfig,
       );
+    } else {
+      // only facilitator config is given.
```

Comment:
It should check if mosaic config `exists`. Because from chain with return blank object if it doesn't exist.

Reply:
I have added suggested approach. 
But i think, **fromChain** should throw error when path is not present.
It will be aligned to what we do in **fromFile** method.
What do you think ?

Reply:
Created ticket to align **fromChain** and **fromFile** methods https://github.com/mosaicdao/mosaic-chains/issues/196